src/shared/sam/cli/samCliBuildCommand.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import { extensionSettingsPrefix } from '../../constants'
+import { fileExists } from '../../filesystemUtilities'
+import { DefaultSettingsConfiguration } from '../../settingsConfiguration'
+import { ChildProcess, ChildProcessResult } from '../../utilities/childProcess'
+import { SamCliCommand } from './samCliCommand'
+import { SamCliConfiguration } from './samCliConfiguration'
+import { DefaultSamCliLocationProvider } from './samCliLocator'
+
+export interface SamCliBuildCommandResponse {
+}
+
+export class SamCliBuildCommand extends SamCliCommand<SamCliBuildCommandResponse> {
+    private readonly _buildDir: string
+    private readonly _baseDir: string
+    private readonly _templatePath: string
+
+    public constructor(
+        buildDir: string,
+        baseDir: string,
+        templatePath: string,
+        config: SamCliConfiguration = new SamCliConfiguration(
+            new DefaultSettingsConfiguration(extensionSettingsPrefix),
+            new DefaultSamCliLocationProvider()
+        )) {
+        super(config)
+        this._buildDir = buildDir
+        this._baseDir = baseDir
+        this._templatePath = templatePath
+    }
+
+    public async execute(): Promise<SamCliBuildCommandResponse> {
+        await this.validate()
+
+        const childProcess: ChildProcess = new ChildProcess(
+            this.samCliLocation!,
+            [
+                'build',
+                '--build-dir', this._buildDir,
+                '--base-dir', this._baseDir,
+                '--template', this._templatePath,
+            ]
+        )
+
+        childProcess.start()
+
+        const childProcessResult: ChildProcessResult = await childProcess.promise()
+
+        if (childProcessResult.exitCode === 0) {
+            const response: SamCliBuildCommandResponse = {}
+
+            return response
+        }
+
+        // tslint:disable-next-line:max-line-length
+        console.error(`SAM CLI error\nExit code: ${childProcessResult.exitCode}\nError: ${childProcessResult.error}\nstdout: ${childProcessResult.stdout}`)
```

Comment:
Template strings [respect newline literals](https://basarat.gitbooks.io/typescript/docs/template-strings.html). Suggest something like:

```typescript
console.error(`SAM CLI error
Exit code: ${childProcessResult.exitCode}
Error: ${childProcessResult.error}
stdout: ${childProcessResult.stdout}`)
```

Or just multiple calls to `console.error`.

Reply:
Changing to multiple calls so that the code remains indented